src/server/debug.rs

```diff
@@ -341,6 +370,90 @@ impl Debugger {
     }
 }
 
+pub struct RegionVerifier {
+    db: Arc<DB>,
+    lock_iter: DBIterator,
+    //    default_iter: DBIterator,
+    write_iter: DBIterator,
+}
+
+impl RegionVerifier {
+    fn new(db: Arc<DB>, reg: Region) -> Result<Self> {
+        let region = reg.clone();
+        let start_key = keys::data_key(region.get_start_key());
+        let end_key = keys::data_end_key(region.get_end_key());
+        let gen_iter = |cf: &str| -> Result<_> {
+            let from = start_key.clone();
+            let to = end_key.clone();
+            let readopts = IterOption::new(Some(from.clone()), Some(to), false).build_read_opts();
+            let handle = box_try!(get_cf_handle(db.as_ref(), cf));
+            let mut iter = DBIterator::new_cf(Arc::clone(&db), handle, readopts);
+            iter.seek(SeekKey::from(from.as_ref()));
+            Ok(iter)
+        };
+
+        Ok(RegionVerifier {
+            db: Arc::clone(&db),
+            write_iter: gen_iter(CF_WRITE)?,
+            lock_iter: gen_iter(CF_LOCK)?,
+            //            default_iter: gen_iter(CF_DEFAULT)?,
+        })
+    }
+
+    pub fn verify_write_by_default(&mut self, wb: &WriteBatch) -> Result<()> {
+        let iter = &mut self.write_iter;
+
+        let write_handle = box_try!(get_cf_handle(&self.db, CF_WRITE));
+        let default_handle = box_try!(get_cf_handle(&self.db, CF_DEFAULT));
+
+        if iter.valid() {
+            loop {
+                let key = iter.key().to_vec();
+                let value = iter.value().to_vec();
+                let write = box_try!(Write::parse(&value));
+                if write.short_value == None {
+                    if let Ok(None) = self.db.get_cf(default_handle, key.as_ref()) {
```

Comment:
seem default and write have the different key format?
Write contains the commit ts, and data contains the start ts, so we can get the data with write format key?

Reply:
ooooops

Reply:
Should truncate commit ts at first,  and then append start ts.

Reply:
yes, it is a big mistake, I think I have deleted keys which should be deleted and should not! Now default cf need to verify too.